### DISCUSSION TITLE
How to make the vscode language pack take effect

### DISCUSSION THREAD

**Discussion by zzkkui** [eclipse-theia/theia#10058] [General] [answered]
about https://github.com/eclipse-theia/theia/pull/9538

> support for vscode language packs to localize Theia itself.

but when I searched and downloaded the language pack in the theia extension and configured this，there seemed to be no change except the place I marked
![image](https://user-images.githubusercontent.com/24466870/132316976-52173018-5528-4fdb-b8b5-38e863c65c3d.png)

Then I tried `i18n sample extension`, and only the commands set in `i18n sample extension` were internationalized. It seems that the vscode language extension is useless?

**Comment by msujew** [comment, marked as answer]
Hi @zzkkui,

my comment about the functionality was a probably a bit misleading. The PR aimed to include the functionality to localize Theia by using language packs, but did not include the actual localization of Theia. This requires quite a lot of work throughout the whole application and I'm currently working on that (no PR so far).

The main idea behind splitting that up was that users of Theia are already able to start translating their custom extensions, while Theia will get its own translation at a later date. I'm currently trying to include it into the 1.18.0 release, but don't quote me on that.

**Reply by zzkkui** [reply]
Oh，Thank you for your answer. 

Now I am working on this requirement. I currently use internationalized dictionaries and `rebind` internal modules, but there are many places where this cannot be achieved. :joy:

I very much hope that the internationalization function will appear in 1.18.0, so I hope to contribute some code. Is there anything i can contribute? :grinning:

**Reply by msujew** [reply]
The translation stuff will take a few days but that's nothing I'm worrying about. However, there is still an open discussion how we handle Theia's custom translation (messages/strings that are not inside of vscode language packs). We are currently looking at using Transifex and machine translations for that. See #9708 for more info on that.

**Comment by chris93111** [comment]
Hi any update ? I have the same issue

**Reply by msujew** [reply]
@chris93111 which version of Theia are you using? Theia has been fully translated for a year now (roughly starting with 1.23) and you shouldn't have to face this issue anymore.

**Reply by chris93111** [reply]
@msujew i use this image quay.io/devspaces/theia-dev-rhel8

**Reply by msujew** [reply]
I have absolutely no idea what version of Theia redhat is using in there.

**Reply by chris93111** [reply]
sorry, they are any files or cmd can help me to find the version of theia used ? it's use che-theia of https://github.com/eclipse-che/che-theia/tree/main/dockerfiles/theia-dev but i not found any version specified in this repository and in container or did i miss something

**Reply by msujew** [reply]
Can you use the `Help->About` menu entry? What does it say?

Alternatively, you will need to find a package.json or something similar that contains references to the `@theia/*` packages. That should tell you the used Theia version. Maybe under `/home/theia-dev`?

**Reply by chris93111** [reply]
CheCode
Version: 1.75.0
Commit: Unknown
Date: 2023-01-15T17:06:02.956Z
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36

**Reply by chris93111** [reply]
{
  "dependencies": {
    "@eclipse-che/theia-generator": "file:/home/theia-dev/eclipse-che-theia-generator",
    "@theia/generator-plugin": "0.0.1-1622834185",
    "generator-code": "^1.7.2",
    "vsce": "^1",
    "yo": "^4.3.1"
  }
}

i thinks is don't use directly theia @theia = {
  "name": "@theia/generator-plugin",
  "version": "0.0.1-1622834185",
  "description": "Yeoman generator for plugins used in Theia IDE",
  "repository": {
    "type": "git",
    "url": "https://github.com/eclipse/theia-generator-plugin/"
